ast/ast.go

```diff
@@ -1766,6 +1804,106 @@ type CreateIndex struct {
 	InterleaveIn *InterleaveIn // optional
 }
 
+// CreateChangeStream is CREATE CHANGE STREAM statement node.
+//
+//	CREATE CHANGE STREAM {{.Name | sql}} {{.For | sql }} {{.Options | sql }}
+type CreateChangeStream struct {
+	// pos = Create
+	// end = Options.end || For.end
+
+	Create  token.Pos // position of "CREATE" keyword
+	Name    *Ident
+	For     ChangeStreamFor
+	Options *ChangeStreamOptions
+}
+
+// ChangeStreamForAll is FOR ALL node in CREATE CHANGE STREAM
+type ChangeStreamForAll struct {
+	// pos = For
+	// end = All
+
+	For token.Pos // position of "FOR" keyword
+	All token.Pos // position of "ALL" keyword
+}
+
+// ChangeStreamForTables is FOR tables node in CREATE CHANGE STREAM
+//
+//	FOR {{.Tables | sqlJoin ","}}
+type ChangeStreamForTables struct {
+	// pos = For
+	// end = Tables[$].end
+
+	For    token.Pos // position of "FOR" keyword
+	Tables []*ChangeStreamForTable
+}
+
+// ChangeStreamForTable table node in CREATE CHANGE STREAM SET FOR
+//
+//	.TableName{{if .Columns}}({{.Columns | sqlJoin ","}}){{end}}
+type ChangeStreamForTable struct {
+	// pos = TableName.pos
+	// end = TableName.end || Rparen + 1
+
+	TableName *Ident
+	Columns   []*Ident
+	Rparen    token.Pos // position of ")"
+}
+
+// ChangeStreamOptions is OPTIONS clause node in CREATE CHANGE STREAM.
+//
+//	OPTIONS ({{.Records | sqlJoin ","}})
+type ChangeStreamOptions struct {
+	// pos = Options
+	// end = Rparen + 1
+
+	Options token.Pos // position of "OPTIONS" keyword
+	Rparen  token.Pos // position of ")"
+
+	Records []*ChangeStreamOptionsRecord
+}
+
+// ChangeStreamOptionsRecord is OPTIONS record node.
+//
+//	{{.Key | sql}}={{.Expr | sql}}
+type ChangeStreamOptionsRecord struct {
+	// pos = Key
+	// end = Value.end
+
+	Key   *Ident
+	Value Expr
+}
+
+// ChangeStreamAlternationSetFor is SET FOR tables node in ALTER CHANGE STREAM
+//
+//	SET FOR {{.For | sql }}
+type ChangeStreamAlternationSetFor struct {
```

Comment:
The name `ChangeStreamAlternationSetFor` seems redundant. `ChangeStreamSetFor` is better. Other `ChangeStreamAlternationXXX` types are ditto.

Reply:
Does this https://github.com/cloudspannerecosystem/memefish/pull/58/files/2107b868eb59e18a8c7d9d6c8f3694e4bf8c7c11#diff-9dcefbf65d58a909b6cc7332b7985d070382bf920a4b13c7acea4ee91ecdf5acR326 interface `ChangeStreamAlternation` need to change name too?

I'm not sure which name is tasteful...

Reply:
At first, I fixed only struct name.
If we also need to fix interface name, please pointing out.

Reply:
You don't need to fix the interface name. `ChangeStreamAlternation` is good.